### ISSUE TITLE
Overwriting JS component trough Plugin

### ISSUE THREAD

**Issue by CompactCodeEU** [amzn/amazon-payments-magento-2-plugin#565]
Your extension is overwriting the shippingAddress JS component trough a plugin. Not sure why it is doing this instead of overwriting it trough the XML fille but it is giving issues with other extensions that also use shippingAddress or try to extend/overwrite the JS component.

**Solution**

Instead of using Amazon\Payment\Plugin\CheckoutProcessor::afterProcess it should use layout xml handling to overwrite the component only on the checkout page.

I know this has been declared really briefly but as a Magento developer this just seems really obvious and i am currently not behind the PC to explain it in detail.

**Comment by tarishah** [comment]
@CompactCodeEU Thanks for reporting this. 

We will look into this and provide an update soon.

**Comment by CompactCodeEU** [comment]
To give some more detail 

So one of our clients has another extension that uses this code 

```
 <item name="shippingAddress" xsi:type="array">
                                                      ..
                                                        <item name="component" xsi:type="string">Cart2Quote_Quotation/js/quote-checkout/view/shipping</item>
                                                      ..
                                                            </item>
```

Which ofcourse is getting overwritten by your extension with a plugin.

**Comment by CompactCodeEU** [comment]
Any update on the matter?

**Comment by tarishah** [comment]
@CompactCodeEU We are currently heads down in Magento 2.3.4 release work. I will have this looked at, by 11/1.

**Comment by CompactCodeEU** [comment]
@tarishah My apologies for the next question. But do you mean 11/1 as in 11th january or 1st Novembre?

**Comment by mrlerch** [comment]
Any update on this from Amazon?

**Comment by mrlerch** [comment]
Sorry, I saw that this will be looked at 11/1. Thanks.

**Comment by CompactCodeEU** [comment]
any updates regarding this topic

**Comment by tarishah** [comment]
@CompactCodeEU We are working through few high priority bugs atm. This ticket is in the queue to be looked at by Friday/early next week. 

Thanks for your patience.

**Comment by mrlerch** [comment]
As it stands, I am unable to use Amazon Login / Amazon Pay, due to it's general incompatibility. As explained it seems to override JS / CSS files after it loads (Your extension is overwriting the shippingAddress JS component trough a plugin). This is causing none of my Quote components, which I need, to work in Magento. I need you to fix this. It's a problem. I have contacted Amazon Support and made them aware that their version 3.2.13 of the Amazon Pay module for M2 is having issues. Thank you.

Sincerely,

Martin

**Comment by christianzichichi** [comment]
Hi @CompactCodeEU  and @mrlerch ,
my apologies for the delay here. This issue was solved after this PR merge (https://github.com/amzn/amazon-payments-magento-2-plugin/pull/594) and included in Amazon Pay plugin 3.3.2 Release. It will also be included in the next Magento releases (starting from M2.3.5).

Thanks.
Regards

Christian
